content/draft/general/keyboard_shortcuts.md

```diff
@@ -0,0 +1,111 @@
+---
+title: "Keyboard shortcuts spec"
+description: ""
+date: 2020-12-01T08:48:57+00:00
+lastmod: 2020-12-01T08:48:57+00:00
+draft: false
+images: []
+menu:
+  raw:
+    parent: "general"
+weight: 100
+toc: true
+---
+
+# Keyboard Shortcuts spec
+
+## Motivation
+
+Users often use keyboard shortcuts to more effectively navigate applications. This spec aims to document the keyboard shortcuts available as well as any preconditions.
+
+## Designs
+
+* [designs](https://www.figma.com/file/IPpvkpDWabBKJTeo6bFop0/Kuba%E2%8E%9CDesktop?node-id=355%3A82493)
+
+## Functional Requirements
+
+### Key definition
+
+The requirements will use the keys `CMD` and `OPT` which are defined for each platform as follows:
+
+- `CMD`
+    - macOS: `⌘` or `CMD`
+    - Windows: `Ctrl`
+    - Linux: `Ctrl`
+- `OPT`
+    - macOS: `⌥` or `OPT`
+    - Windows: `Alt`
+    - Linux: `Alt`
+
+### Keyboard shortcuts help
+
+- `CMD + /` MUST open a modal listing available key shortcuts
+- The Modal listing available keyshortcuts MUST display the correct keys depending on the platform, for example `CMD + C` on macOS and `Ctrl + C` on Windows.
+
+### Keyboard shortcuts
+
+#### Message focus
+
+- `TAB` - MUST trigger focus of the last message
+- `UP` - MUST focus on the previous message
+    - requirement: there is already a message in focus
+- `DOWN` - MUST focus on the previous message
+    - requirement: there is already a message in focus
+- `ESCAPE`:
+    - if a message is already in focus, it MUST be removed from focus and the input box MUST be focused instead
+    - if input box is already in focus then focus MUST BE removed completly
+    - if there is nothing in focus then the input box MUST be focused
+
+#### Message selection shortcuts
+
+requirement: there is already a message in focus
+
+- `E` MUST edit the currently focused message
+    - requirement: the message must have been sent by the current user
+- `Backspace` MUST delete the currently focused message
+    - requirement: the message must have been sent by the current user
+- `P` pins the currently focused message
+    - requirement: the current user must be an admin in a community or a group chat
+- `+` react to currently focused message
```

Comment:
this shortcut is currently unclear, because we react to messages by selecting one of the 4 possible emojis. So either the shortcut needs to be expanded like `+` + `1` selects the first reaction emoji for example, OR the application needs to be updated so it can support any emoji as a reaction, and this would trigger instead the emoji picker.